src/lib/components/custom/component-template.svelte

```diff
@@ -0,0 +1,132 @@
+<script lang="ts">
+	import Typography from '$lib/components/ui/typography.svelte'
+	import VerticalContainer from '$lib/components/custom/vertical-container.svelte'
+	import type { Snippet } from 'svelte'
+
+	type Props = {
+		name: string
+		tagline: string
+		description: Snippet
+		examples: Snippet
+		controls: Snippet
+		preview: Snippet
+		implement: Snippet
+		choose?: Snippet
+		sizes?: Snippet
+	}
+
+	const {
+		name,
+		tagline,
+		description,
+		examples,
+		controls,
+		preview,
+		implement,
+		choose,
+		sizes,
+	}: Props = $props()
+</script>
+
+<section>
+	<Typography variant="h2" bold>{name}</Typography>
+	<Typography variant="large">
+		{tagline}
+	</Typography>
+</section>
+
+<VerticalContainer>
+	<section class="description">
+		{@render description()}
+	</section>
+	<section class="examples">
+		{@render examples()}
+	</section>
+</VerticalContainer>
+
+<hr />
+
+<section id="use">
+	<Typography class="title" variant="h4" bold>Use</Typography>
+	<div class="title" />
+
+	<VerticalContainer>
+		<section class="controls">
+			{@render controls()}
+		</section>
+
+		<section class="preview">
+			{@render preview()}
+		</section>
+	</VerticalContainer>
+</section>
+
+<hr />
+
+<section id="implement">
+	<Typography class="title" variant="h4" bold>Implement</Typography>
+	<div class="title" />
+	{@render implement()}
+</section>
+
+<hr />
+
+{#if choose}
+	<section id="choose-a-button-type">
+		<Typography class="title" variant="h4" bold>Choose a button type</Typography>
+		<div class="title" />
+		{@render choose()}
+	</section>
+
+	<hr />
+{/if}
+
+{#if sizes}
+	<section id="about-sizes">
+		<Typography class="title" variant="h4" bold>About sizes</Typography>
+		<div class="title" />
+		{@render sizes()}
+	</section>
+
+	<hr />
+{/if}
+
+<section id="footer">
+	<Typography>
+		Made by diete. Source code is available on <a
+			href="https://github.com/diete-design/diete.design">Github</a
+		>.
+	</Typography>
+</section>
```

Comment:
Please move this to `+layout.svelte` which is either in the `routes/components` folder or in root